packages/core/src/parser/cstToAstTransformer.ts

```diff
@@ -276,17 +268,70 @@ export function generateCstToAstTransfromer(parser: Parser): ICstVisitor<never,
         }
 
         /**
-         * Maps a call expression to an Expression
+         * Maps a simple call expression to a generator for an Expression
+         * The generated expression depends on whether there are any call brackets,
+         * and whether the expression is a field access or an index expression.
          *
          * @param ctx the children of the current CST node
-         * @returns a definition of the field access
+         * @returns a generator for the expression
          */
-        private simpleCallExpression(ctx: any): AccessDefinition {
-            const token = ctx.Identifier[0];
-            return {
-                identifier: token.image,
-                identifierRange: generateRange(token, token),
-                callBrackets: (ctx.callBrackets ?? []).map((brackets: CstNode) => this.visit(brackets))
+        private simpleCallExpression(ctx: any): SimpleCallExpressionGenerator {
+            const callBrackets = (ctx.callBrackets ?? []).map((brackets: CstNode) => this.visit(brackets));
+            let identifier: IToken | undefined = undefined;
+            let index: Expression | undefined = undefined;
+            if (ctx.expression) {
+                index = this.visit(ctx.expression[0]);
+            } else {
+                identifier = ctx.Identifier[0];
+            }
+            return (baseExpression) => {
+                const startPos = baseExpression.range;
+                if (identifier != undefined) {
```

Comment:
Method is too long and does too much.
Extract submethods.

Reply:
I knew I forgot sth
c0f53c49eac7c2f07875ca478d4e9f3647a3502d